### ISSUE TITLE
CFNetwork warning when iterating over ChatChannel.latestMessages

### ISSUE THREAD

**Issue by AndrewSB** [GetStream/stream-chat-swiftui#194]
Hey martin! noticed this purple warning from Xcode you might like to be aware of

<img width="1014" alt="Screen Shot 2022-09-06 at 4 05 03 PM" src="https://user-images.githubusercontent.com/3814772/188755319-d57bc135-e158-4679-8ec2-a1d697878181.png">

Xcode put the breadcrumb on this expression:

```
extension ChatChannel { 
...
        return latestMessages.filter { message in
            !message.isSentByCurrentUser && !message.isDeleted && message.createdAt > myLastRead
        }
```

**Comment by AndrewSB** [comment]
as an update: this might be urgent. it blocks the main thread and could cause a watchdog crash

the file that's being downloaded is the JSON for our custom attachment

**Comment by martinmitrevski** [comment]
Hey @AndrewSB,

I've analysed this, and it's actually in your codebase, specifically in the `chatMessage.reclipAttachment` computed variable.

There, you are using `let playbackInfoData = try Data(contentsOf: playbackInfoFileURL)`. This method is synchronous, so if it's called from the main thread, it will block it.

For example, I've found a place where you're calling this on the main thread, in your SwiftUI views:
```swift
if unreadMessagesCount > 0 {
     Image(
          channel.myUnreadMessages.filter { !$0.isSentByCurrentUser }.contains(where: { $0.reclipAttachment != nil }) ? "IconChatUnviewedReclip" : "IconChat"
    )
    .frame(width: Grid.x4, height: Grid.x4)
}
```
If that's called before the attachment is loaded in your other async loading code, it will be invoked from the main thread and cause this warning. And the warning you get corresponds with the `channel.myUnreadMessages.filter` call in this view.

Since there's no guarantee about this (it can happen even in 4.20.0), we recommend to make your code async (e.g. using Futures, as you are already doing it in several places), or update the logic in the SwiftUI view, not to use reclipAttachment (and also check other places).

Hope this helps,
Martin
